ml_experiment/Scheduler.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+import sys
+import os
+sys.path.append(os.getcwd())
+
+import sqlite3
+from typing import Self, Callable, NamedTuple
+from dataclasses import dataclass
+from itertools import product
+from multiprocessing.pool import Pool
+import subprocess
+from ml_experiment.metadata.MetadataTableRegistry import MetadataTableRegistry
+
+class RunSpec(NamedTuple):
+    part_name: str
+    version: int
+    config_id: int
+    seed: int
+
+
+@dataclass
+class RunConfig:
+    ...
```

Comment:
This commit does not contain the slurm functionality but the parent `RunConfig` is still defined here. I guess this is fine?

Reply:
Yup, I think that's just fine!

I typically try to avoid defining generic interfaces before they are needed, but in this case we _know_ that this will be needed in a follow-up PR.